model/task_queue.go

```diff
@@ -471,7 +475,21 @@ type taskQueueQuery struct {
 	DistroID   string
 }
 
-func findTaskQueueForDistro(q taskQueueQuery) (*TaskQueue, error) {
+func getTaskQueueFilter(taskQueueItemKey, value string) bson.M {
```

Comment:
Is the purpose of this to only return results which match the filter? I would think that you want to do a $match here instead, though it may vary depending on what the pipeline looks like before this

Reply:
it's for filtering the queue array for items that match the filters

Reply:
Then I think you want to do a $match. $project is used to remove/recompute fields

Reply:
that's not what the mongo docs say and this works

Reply:
https://docs.mongodb.com/manual/reference/operator/aggregation/filter/

Reply:
I'm talking about the $project stage, not the operator you used within that stage. Normally you'd use $project to do field-level operations like remove/add fields or change the data within an existing one, but unless I'm missing something you're trying to filter out documents that don't match the task ID. It might work, but it's written confusingly

Reply:
the $project stage is an example of filtering an array that i took directly from the mongo docs, so i struggle to understand why it can't work here, especially since it does work. 

Per the docs, which i linked above: 

> The following example filters the items array to only include documents that have a price greater than or equal to 100:

```
db.sales.aggregate([
   {
      $project: {
         items: {
            $filter: {
               input: "$items",
               as: "item",
               cond: { $gte: [ "$$item.price", 100 ] }
            }
         }
      }
   }
])
```

If you think it's necessary to block the PR because of this approach then please provide an example of what you feel is the correct way.